tests/TestHarness/Node.py

```diff
@@ -273,6 +273,54 @@ def isProducer():
             f"Waited for {time.perf_counter()-start} sec but never found producer: {producer}. Started with {initialProducer} and ended with {self.getInfo()['head_block_producer']}"
         return found
 
+    # returns True if the node has paused production.
+    def paused(self):
+        # Cannot use producer_plugin's paused() endpoint as it does not
+        # include paused due to max-reversible-blocks exceeded.
+        # The idea is to find the scheduled start block of node's producer's
+        # next round. If that block is not produced, it means block production
+        # on the node is paused.
+
+        scheduled_producers = []
+        schedule = self.processUrllibRequest("chain", "get_producer_schedule")
+        for prod in schedule["payload"]["active"]["producers"]:
+            scheduled_producers.append(prod["producer_name"])
+        if Utils.Debug: Utils.Print(f'scheduled_producers {scheduled_producers}')
+
+        self.getInfo()
+        currBlockNum=self.lastRetrievedHeadBlockNum
+        currProducer=self.lastRetrievedHeadBlockProducer
+        blocksRemainedInCurrRound = 12 - currBlockNum % 12 - 1
+        if Utils.Debug: Utils.Print(f'currBlockNum {currBlockNum}, currProducer {currProducer}, blocksRemainedInCurrRound {blocksRemainedInCurrRound}')
+
+        # find the positions of currProducerPos and nodeProducer in the schedule
+        currProducerPos=0
+        nodeProducerPos=0
+        for i in range(0, len(scheduled_producers)):
+            if scheduled_producers[i] == currProducer:
+                currProducerPos=i
+            if scheduled_producers[i] == self.producerName:
+                nodeProducerPos=i
+
+        # find the number of the blocks to node producer's next scheduled round
+        blocksToNextScheduledRound = 0
+        if currProducerPos < nodeProducerPos:
+            # nodeProducerPos - currProducerPos - 1 is the number of producers
+            # from current producer to the node producer in the schedule
+            blocksToNextScheduledRound = (nodeProducerPos - currProducerPos - 1) * 12 + blocksRemainedInCurrRound + 1
```

Comment:
Assumes 12 blocks per round. Many tests do assume 12 blocks per round, but would be nice to stop assuming that is always the case. At the very least make it a const at top of function.